### DISCUSSION TITLE
Nx autocomplete

### DISCUSSION THREAD

**Discussion by Tiedye** [nrwl/nx#23278] [Feature Requests]
## Description

Follow up to #6151

> It would be decent to be able to use autocomplete when typing nx commands
> 
> Im thinking about two ways
> 
>     make the - - help compliant with compdef _gnu_generic nx
> 
>     create completions through bash and zsh
> 
>     using completion npm packages in the cli
> 

- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Motivation

It would make it a lot more pleasant to use when used from the cli instead of with an extension.

## Suggested Implementation

Someone already did it for [zsh](https://github.com/jscutlery/nx-completion), I haven't looked at the implementation but at least its possible

**Comment by FrozenPandaz** [comment]
I've converted this to a discussion so this can continue to be designed out.
